codec.go

```diff
@@ -97,15 +161,25 @@ func decodeResponse(data []byte) (Message, error) {
 // decodeEvent determines what event type in the ProtocolMessage hierarchy
 // data corresponds to and uses json.Unmarshal to populate the corresponding
 // struct to be returned.
-func decodeEvent(data []byte) (Message, error) {
+func (c *Codec) decodeEvent(data []byte) (Message, error) {
 	var e Event
 	if err := json.Unmarshal(data, &e); err != nil {
 		return nil, err
 	}
-	if ctor, ok := eventCtor[e.Event]; ok {
+	if ctor, ok := c.eventCtor[e.Event]; ok {
 		eventPtr := ctor()
 		err := json.Unmarshal(data, eventPtr)
 		return eventPtr, err
 	}
 	return nil, &DecodeProtocolMessageFieldError{e.GetSeq(), "Event", "event", e.Event}
 }
+
+// DecodeProtocolMessage parses the JSON-encoded data and returns the result of
```

Comment:
Note that this doc comment is repeated almost verbatim above.

Reply:
nit: How about // DecodeProtocolMessage parses the JSON-encoded ProtocolMessage and returns the message embedded in it? I hope that makes it clearer what this expects 'data' to be.